crates/api_common/src/request.rs

```diff
@@ -110,16 +111,20 @@ pub fn generate_post_link_metadata(
       &context,
     )
     .await;
-    if let Some(thumbnail_url_) = metadata.thumbnail {
-      thumbnail_url = Some(thumbnail_url_.into());
```

Comment:
This would overwrite any custom thumbnail given, with one fetched from the metadata.